plugin/impl.go

```diff
@@ -90,178 +84,126 @@ func (p *Plugin) Execute() error {
-							}
-						}
-						if build == nil {
-							return fmt.Errorf("unable to get last successful build for %s", entry)
-						}
-					} else {
-						// Get build by number
-						buildNumber, _ := strconv.Atoi(branch)
-						build, err = client.Build(owner, name, buildNumber)
-						if err != nil {
-							return fmt.Errorf("unable to get requested build %v for deploy for %s", buildNumber, entry)
-						}
-					}
-					if p.settings.Wait && !waiting && (build.Status == drone.StatusRunning || build.Status == drone.StatusPending) {
-						fmt.Printf("BuildLast for repository: %s, returned build number: %v with a status of %s. Will retry for %v.\n", entry, build.Number, build.Status, p.settings.Timeout)
-						waiting = true
-						continue
-					}
-					if (build.Status != drone.StatusRunning && build.Status != drone.StatusPending) || !p.settings.Wait {
-						// start a new deploy
-						var newBuild *drone.Build
-						newBuild, err = client.Promote(owner, name, int(build.Number), p.settings.Deploy, p.settings.params)
-						if err != nil {
-							if waiting {
-								continue
-							}
-							return fmt.Errorf("unable to trigger deploy for %s - err %v", entry, err)
-						}
-						fmt.Printf("starting deploy for %s/%s env - %s build - %d\n", owner, name, p.settings.Deploy, build.Number)
-						logParams(p.settings.params, p.settings.ParamsEnv.Value())
-
-						if p.settings.Block {
-							err = blockUntilBuildIsFinished(p, client, owner, name, int(newBuild.Number))
-							if err != nil {
-								return err
-							}
-						}
-
-						break I
-					}
-				}
+		build, err = client.Build(owner, name, buildNumber)
+		if err != nil {
+			return fmt.Errorf("unable to get requested build %d for deploy for %s", buildNumber, entry)
+		}
+	}
 
-				// get the latest build for the specified repository
-				build, err := client.BuildLast(owner, name, branch)
-				if err != nil {
-					if waiting {
-						continue
-					}
-					return fmt.Errorf("unable to get latest build for %s: %s", entry, err)
-				}
-				if p.settings.Wait && !waiting && (build.Status == drone.StatusRunning || build.Status == drone.StatusPending) {
-					fmt.Printf("BuildLast for repository: %s, returned build number: %v with a status of %s. Will retry for %v.\n", entry, build.Number, build.Status, p.settings.Timeout)
-					waiting = true
-					continue
-				} else if p.settings.LastSuccessful && build.Status != drone.StatusPassing {
-					builds, err := client.BuildList(owner, name, drone.ListOptions{})
-					if err != nil {
-						return fmt.Errorf("unable to get build list for %s", entry)
-					}
-
-					build = nil
-					for _, b := range builds {
-						if b.Source == branch && b.Status == drone.StatusPassing {
-							build = b
-							break
-						}
-					}
-					if build == nil {
-						return fmt.Errorf("unable to get last successful build for %s", entry)
-					}
-				}
+	fmt.Printf("deploying Drone build #%d of repo %s/%s\n", build.Number, owner, name)
```

Comment:
@bradrydzewski Question: The old code, if the _wait_ setting is enabled, will delay deploying/promoting a build until it has finished running. Considering you think the wait option is legacy, what should the new behaviour be?